src/devices/sound/nes_apu.cpp

```diff
@@ -143,50 +143,96 @@ void nesapu_device::device_start()
 
 	calculate_rates();
 
+	// calculate mixer output
+	/*
+	pulse channel output:
+
+		     95.88
+	-----------------------
+	      8128
+	----------------- + 100
+	pulse 1 + pulse 2
+
+	*/
+	for (int i = 0; i < 31; i++)
+	{
+		stream_buffer::sample_t pulse_out = (i == 0) ? 0.0 : 95.88 / ((8128.0 / i) + 100.0);
+		m_square_lut[i] = pulse_out;
+	}
+
+	/*
+	triangle, noise, DMC channel output:
+
+		         159.79
+	-------------------------------
+	            1
+	------------------------- + 100
+	triangle   noise    dmc
+	-------- + ----- + -----
+	  8227     12241   22638
+
+	*/
+	for (int t = 0; t < 16; t++)
+	{
+		for (int n = 0; n < 16; n++)
+		{
+			for (int d = 0; d < 128; d++)
+			{
+				stream_buffer::sample_t tri_out = (t == 0) ? 0.0 : (t / 8227.0);
+				stream_buffer::sample_t noise_out = (n == 0) ? 0.0 : (n / 12241.0);
+				stream_buffer::sample_t dmc_out = (d == 0) ? 0.0 : (d / 22638.0);
+				stream_buffer::sample_t tnd_out = tri_out + noise_out + dmc_out;
```

Comment:
Can't this just be `stream_buffer::sample_t tnd_out = t/8227.0 + n/12241.0 + d/22638.0;` or what not on a single line? In other words the single test of `tnd_out == 0.0` on the next line should suffice, no? Or if we were being hyper defensive about float comparison it might be wise to just check `t + n + d == 0`, not that it should matter with these known values.

Anyway, this addresses my previous concerns otherwise. Formulas look great!